client/components/list-item.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+
+export default function ListItem(props) {
+
+  const [isActive, toggleActive] = useState(props.dateInfo.isActive);
+  let imgURL = '../../server/images/checked-box.svg';
+  let imgAlt = 'checked';
+  let strikeClass = '';
+  if (isActive) {
+    imgURL = '/images/checked-box.svg';
+    imgAlt = 'checked';
+  } else {
+    imgURL = '/images/unchecked-box.svg';
+    imgAlt = 'unchecked';
+    strikeClass = ' strike';
+  }
+
+  const setIsActive = () => {
+    const req = {
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' }
+    };
+    fetch(`/api/dateActive/${props.dateInfo.dateId}`, req)
+      .then(toggleActive(!isActive))
```

Comment:
Careful, this isn't actually waiting until the promise is fulfilled. Be sure to pass `.then()` a function definition rather than a call.

```js
.then(() => toggleActive(!isActive))
```